compiler/src/Language/Bond/Codegen/Cpp/Types_cpp.hs

```diff
@@ -99,3 +102,7 @@ types_cpp cpp file _imports declarations = ("_types.cpp", [lt|
     } // namespace _bond_enumerators|]
 
     statics _ = mempty
+
+    anyEnum [] = False
```

Comment:
There's [`isEnum`](https://github.com/Microsoft/bond/blob/939c298ff6d619dbc0bc9f5ca25c2088de86d83b/compiler/src/Language/Bond/Syntax/Util.hs#L132-L135) in Utils.hs. It handles aliases as well.

With that and [`any`](http://hackage.haskell.org/package/base-4.10.0.0/docs/Prelude.html#v:any)
```
anyEnum :: [Type] -> Bool
anyEnum = any isEnum
```
or, alternatively if you just need a value
```
hasEnumDecl :: Bool
hasEnumDecl = any isEnum declarations
```
or just do it all together:

```
 unorderedMapInclude = if (not (any isEnum declarations)) then mempty else [lt|#include <unordered_map>
|lt]
```

Reply:
Here we care about declarations and not types, so cannot use `isEnum` you mentioned and we don't care about aliases. I added `isEnumDelcaration` and will use `any` (thanks!).